aws/resource_aws_waf_rate_based_rule_test.go

```diff
@@ -156,6 +156,58 @@ func TestAccAWSWafRateBasedRule_changePredicates(t *testing.T) {
 	})
 }
 
+// Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/9659
+func TestAccAWSWafRateBasedRule_changeRateLimit(t *testing.T) {
+	var ipset waf.IPSet
+	var before, after waf.RateBasedRule
+	var idx int
+	ruleName := fmt.Sprintf("wafrule%s", acctest.RandString(5))
+	resourceName := "aws_waf_rate_based_rule.wafrule"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSWaf(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafRuleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSWafRateBasedRuleConfig_changeRateLimit(ruleName, 4000),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAWSWafIPSetExists("aws_waf_ipset.ipset", &ipset),
+					testAccCheckAWSWafRateBasedRuleExists(resourceName, &before),
+					resource.TestCheckResourceAttr(resourceName, "name", ruleName),
+					resource.TestCheckResourceAttr(resourceName, "rate_limit", "4000"),
+					resource.TestCheckResourceAttr(resourceName, "predicates.#", "1"),
+					computeWafRateBasedRulePredicateWithIpSet(&ipset, false, "IPMatch", &idx),
```

Comment:
For follow up: this looks like another one of those test functions to determine TypeSet index which can probably be removed with the awesome new functions. 👍